app/uk/gov/hmrc/exports/mongock/changesets/CacheChangeLog.scala

```diff
@@ -68,4 +67,80 @@ class CacheChangeLog {
 
     service.allCountries.find(_.countryName == countryName).map(_.countryCode).getOrElse(countryName)
   }
+
+  @ChangeSet(order = "003", id = "CEDS-2247 Change origination country structure", author = "Patryk Rudnicki")
+  def changeOriginationCountryStructure(db: MongoDatabase): Unit = {
+    val queryParams: ImmutableMap[String, AnyRef] =
+      ImmutableMap.of("locations.originationCountry", ImmutableMap.of("$exists", true, "$ne", ""))
+    val query: Document = new Document(queryParams)
+
+    val documents = db.getCollection(collection).find(new BasicDBObject(query)).toSeq
+
+    documents.foreach { document =>
+      val locations = document.get("locations", classOf[Map[String, String]])
+      val originationCountry: String = locations.get("originationCountry")
+
+      val queryIndexes: Map[String, String] =
+        ImmutableMap.of("id", document.get("id").asInstanceOf[String], "eori", document.get("eori").asInstanceOf[String])
```

Comment:
Why are we referencing "eori" here?

Reply:
Are id and eori keys to the document?

Reply:
yeah, we treated this like identifier to easier find the documents.

Implemented with @PauloPortugal

Reply:
Thought so.  Maybe we could extract that bit out as we will need it whenever we update a declaration?

Reply:
I tried this, but it's related with the information from specific document. I can extract this as private method if it will be more readable or helpful.